### ISSUE TITLE
Nested arrays parsed incorrectly

### ISSUE THREAD

**Issue by sinitsyn-alex** [ljharb/qs#288]
Hi,
I have this object:
```
const filter = {
  3: [{ name: 'Test', id: 50753 }, { name: 'Test2', id: 48464 }],
  6: ['2018-11-01', '2018-11-02']
}
```
I call stringify and parse
```
const stringifyQuery = Qs.stringify(filter, { arrayFormat: 'brackets' })
const parseQuery = Qs.parse(stringifyQuery, { arrayLimit: 0 })
```

Result parse:
Key '3' contains only 1 element instead of 2.
And this element contains the merged keys `name, id`:
```
3: Array(1)
  0:
    id: (2) ["50753", "48464"]
    name: (2) ["Test", "Test2"]
```

I expect:
```
3: Array(2)
  0: {name: "Test", id: 50753}
  1: {name: "Test2", id: 48464}
```

Please help me
https://codepen.io/anon/pen/rQrqXx

**Comment by ljharb** [comment]
I believe this is either a duplicate of, or related to, #252 and #215. It's also possible this was recently fixed in #242.

**Comment by ljharb** [comment]
Although - `qs.stringify(filter, { arrayFormat: 'brackets', encodeValuesOnly: true })` produces `'3[][name]=Test&3[][id]=50753&3[][name]=Test2&3[][id]=48464&6[]=2018-11-01&6[]=2018-11-02'` - and as you can see, they're using `[]`, without indices. I'd expect, thus, this to parse to `3` being an array with three items - ie, `{ 3: [{ name: 'Test' }, { id: '50753' }, { id: 48464 }], 6: blah }` - so this seems like an issue with `parse` not supporting a "brackets" arrayFormat option.

See https://github.com/ljharb/qs/issues/252#issuecomment-370064658 specifically.

**Comment by tfwright** [comment]
I don't think this is (just) an issue with parsing. I ran into this today using a Ruby library to parse the query string that `stringify` produces. It works fine in most cases with objects and arrays, but in some cases it seems to merge objects in an array into _keys_ on a single object in the array.

`qs.stringify({test: [{ array: [{obj: 1}]}, {obj: 2}]}, { arrayFormat: 'brackets' }))`

produces

`"test[][array][][obj]=1&test[][obj]=2"`

which is parsed as

`{"test":[{"array":[{"obj":"1"}],"obj":"2"}]}`

with both the Ruby parser and qs.parse.

However, I note that the Ruby stringifier produces the exact same query string, so perhaps this is an intrinsic limitation of the format?
